### PR TITLE
New backend rounded corner, part 1

### PR DESCRIPTION
**Pull request by yshui**
This pull request add basic rounded corner for the new glx and xrender backends.

### CODE REVIEW

**Comment by tryone144** [comment]
>  Hi, does this look reasonable to you?

I'll try to find some time over the weekend to take a look at this.

File: src/backend/xrender/xrender.c [hunk 1]
```diff
@@ -382,10 +470,11 @@ static void release_image_inner(backend_t *base, struct _xrender_image_data_inne
...
 static void release_image(backend_t *base, void *image) {
-	struct backend_image *img = image;
-	img->inner->refcount--;
-	if (img->inner->refcount == 0) {
-		release_image_inner(base, (void *)img->inner);
+	struct xrender_image *img = image;
+	xcb_free_pixmap(base->c, img->rounded_rectangle);
+	img->rounded_rectangle = XCB_NONE;
+	if (img->base.inner->refcount == 0) {
+		release_image_inner(base, (void *)img->base.inner);
```
Comment (tryone144) [inline src/backend/xrender/xrender.c:477]: Is it correct, that we don't reduce the `refcount` of the inner image now? We did so previously.

**Review by tryone144 — APPROVED** [review]
Look good to me. :+1:
I just think the ref-counting got shuffled up while adding the cached mask image to the xrender backend.

File: src/backend/backend.c [hunk 2]
```diff
@@ -339,6 +339,9 @@ void paint_all_new(session_t *ps, struct managed_win *t, bool ignore_damage) {
 			    &dim_opacity);
 			ps->backend_data->ops->set_image_property(
 			    ps->backend_data, IMAGE_PROPERTY_OPACITY, w->win_image, &w->opacity);
+			ps->backend_data->ops->set_image_property(
+			    ps->backend_data, IMAGE_PROPERTY_CORNER_RADIUS, w->win_image,
+			    (double[]){w->corner_radius});
```
[the remark this answers is not in the document]
Comment (yshui) [reply src/backend/backend.c:344]: This is specifically done here to cast `corner_radius` to `double`.

File: src/backend/xrender/xrender.c (hunk 1 above)
Comment (yshui) [reply src/backend/xrender/xrender.c:477]: Nice catch!

**Comment by yshui** [comment]
@tryone144 thanks for the review! looks like this is good to be merged.

**Pull request merged**